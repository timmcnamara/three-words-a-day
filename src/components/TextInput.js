import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { dark } from "../styles/styleGuide";

const InputStyled = styled.input`
  background: white;
  color: ${dark};
  font-size: 1em;
  margin: 10px;
  padding: 0.75em 1em;
  border: 2px solid ${dark};
  border-radius: 3px;
  min-width: 250px;
`;

const TextInput = ({ placeholder, onChange }) => {
  return (
    <InputStyled
      aria-label="input"
      placeholder={placeholder}
      onChange={onChange}
    />
  );
};

TextInput.propTypes = {
  placeholder: PropTypes.string,
  onChange: PropTypes.func
};

export default TextInput;
