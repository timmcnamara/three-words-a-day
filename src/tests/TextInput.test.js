import { render, cleanup, fireEvent } from "react-testing-library";
import React from "react";
import TextInput from "../components/TextInput";
import "jest-dom/extend-expect";

const props = {
  placeholder: "placeholder",
  onChange: jest.fn()
};

afterEach(cleanup);

it("render TextInput component", () => {
  const { asFragment } = render(<TextInput />);
  // run a snapshot
  expect(asFragment()).toMatchSnapshot();
});

it("renders the placeholder text", () => {
  const { getByPlaceholderText } = render(<TextInput {...props} />);
  const { placeholder } = getByPlaceholderText("placeholder");
  expect(placeholder).toBe("placeholder");
});

it("Calls onChange function on change", () => {
  const { getByLabelText } = render(<TextInput {...props} />);
  fireEvent.change(getByLabelText("input"), { target: { value: "update" } });
  expect(props.onChange).toBeCalled();
});
