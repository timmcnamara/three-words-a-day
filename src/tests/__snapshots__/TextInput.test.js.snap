// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render TextInput component 1`] = `
<DocumentFragment>
  <input
    aria-label="input"
    class="sc-bdVaJa jhJpaq"
  />
</DocumentFragment>
`;
